docs/developer_guide.md

```diff
@@ -0,0 +1,59 @@
+# Setup Envoy AI Gateway Local Development Environment
+
+## Prerequisites
+- Go 1.23 or later
+- Envoy proxy installed
+- OpenAI API or AWS Bedrock key
+
+## Build Required Binaries
+
+```bash
+git clone https://github.com/envoyproxy/ai-gateway
+cd ai-gateway
+
+# Build the extproc binary
+go build -o out/extproc cmd/extproc/main.go
+```
+
+## Start Envoy
+
+Download Envoy from [release page](https://github.com/envoyproxy/envoy/releases). 
```

Comment:
This will not work for mac users as envoy doesn't publish binary. you might want to refer to Envoy doc instead of github releases.